app/client/components/delivery/address/formValidation.tsx

```diff
@@ -0,0 +1,402 @@
+import { DeliveryAddress } from "../../../../shared/productResponse";
+import { ukPhoneNumberWithoutPrefix } from "../../callCentreNumbers";
+
+interface ErrorState {
+  isValid: boolean;
+  message: string;
+}
+
+export interface FormValidationResponse {
+  isValid: boolean;
+  addressLine1?: ErrorState;
+  town?: ErrorState;
+  postcode?: ErrorState;
+  country?: ErrorState;
+}
+
+export const isFormValid = (
+  formData: DeliveryAddress,
+  subscriptionsNames: string[]
+): FormValidationResponse => {
+  const addressLine1 = {
+    isValid: formData.addressLine1.length > 0,
+    message: "Please enter an address"
+  };
+  const town = {
+    isValid: !!(formData.town && formData.town.length > 0),
+    message: "Please enter a city"
```

Comment:
is there a reason for mixing _town_ and _city_ in naming?

Reply:
nope :s ... I've checked with Jess. It should be Town/City